client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+import os
+import sys
+import time
+import stat
+import shutil
+
+import king_phisher.client.plugins as plugins
+
+from gi.repository import Gtk
+from gi.repository.GdkPixbuf import Pixbuf
+
+import boltons.strutils
+import boltons.timeutils
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Josh Jacob']
+	title = 'SFTP Client GUI'
+	description = """
+	How many foos could fu foo is fu fooed fu foos on foo.
+	"""
+	homepage = 'https://github.com/securestate/king-phisher'
+
+	def initialize(self):
+		module = sys.modules['king_phisher.client.application']
+		self.temp = module.KingPhisherClientApplication.do_sftp_client_start
+		module.KingPhisherClientApplication.do_sftp_client_start = self.new_SFTP_signal
+		sys.modules['king_phisher.client.application'] = module
+		return True
+
+	def _cleanup(self):
+		module = sys.modules['king_phisher.client.application']
+		module.KingPhisherClientApplication.do_sftp_client_start = self.temp
+		sys.modules['king_phisher.client.application'] = module
+
+	def new_SFTP_signal(self):
+		if not self.application._ssh_forwarder:
+			config = self.application.config
+			server = config['server']
+			server = server.split(':')
+			username = config['server_username']
+			password = config['server_password']
+			self.application._create_ssh_forwarder(server, username, password)
+		connection = self.application._ssh_forwarder
+		ssh = connection.client
+		ftp = ssh.open_sftp()
+
+		target_file = os.path.splitext(__file__)[0]
+		builder = Gtk.Builder()
+		builder.add_from_file(target_file + '.ui')
+		self.window = builder.get_object('SFTPClientGUI.window')
+		manager = FileManager(builder, self.application, ftp)  #pylint: disable=unused-variable
```

Comment:
Instead of being passed the builder, since the `Plugin` class doesn't need it, the `FileManager` class should create it itself. The `Plugin.window` attribute should then be set from the `FileManager.window` or accessed as a property.